crates/core/src/test_utils/deploy_helpers.rs

```diff
@@ -536,6 +536,35 @@ impl DeployedKakarot {
     }
 }
 
+/// Returns a `StarknetConfig` instance customized for Kakarot.
+pub fn get_kakarot_starknet_config() -> StarknetConfig {
+    let kakarot_steps = 2u32.pow(24);
+    StarknetConfig {
+        allow_zero_max_fee: true,
+        env: Environment {
+            chain_id: "SN_GOERLI".into(),
+            invoke_max_steps: kakarot_steps,
+            validate_max_steps: kakarot_steps,
+            ..Default::default()
+        },
+        ..Default::default()
+    }
+}
+
+/// Constructs a test sequencer with the Starknet configuration tailored for Kakarot.
+///
+/// This function initializes a `TestSequencer` instance with the default `SequencerConfig`
+/// and a custom `StarknetConfig` obtained from the `get_kakarot_starknet_config()` function.
+/// The custom `StarknetConfig` sets the chain_id to "SN_GOERLI" and both the `invoke_max_steps`
+/// and `validate_max_steps` to `2**24`. It also sets `allow_zero_max_fee` to true in
+/// `StarknetConfig`. This setup is aimed to provide an appropriate environment for testing
+/// Kakarot based applications.
+///
+/// Returns a `TestSequencer` configured for Kakarot.
+pub async fn construct_kakarot_test_sequencer() -> TestSequencer {
```

Comment:
@greged93, does that look like a nice opportunity to have a builder struct?
otherwise, lgtm

Reply:
still don't have intuition of when the builder pattern is useful over when you have a smol number of params, it will come to me in a dream